browser/src/control/Control.UserList.ts

```diff
@@ -434,6 +461,22 @@ class UserList extends L.Control {
 
 		(document.getElementById('follow-editor-checkbox') as HTMLInputElement).checked = this.map._docLayer._followEditor;
 	}
+
+	renderFollowingChip() {
+		const followingChip = document.getElementById('followingChip');
+
+		const following = this.getFollowedUser();
+
+		if (following === undefined || !this.map._docLayer._followUser) {
+			followingChip.style.display = 'none';
+			return;
+		}
+
+		followingChip.innerHTML = this.options.followingChipText.replace('%user', following[1].username);
```

Comment:
can we not use innerHTML but create nodes and use innerText? (username is something what comes from user/integration input)

Reply:
Sure, that makes sense

We directly insert the username as HTML in the user join/leave messages

```ts
		if (type === 'join') {
			message = this.options.userJoinedPopupMessage.replace('%user', username);
		} else {
			message = this.options.userLeftPopupMessage.replace('%user', username);
		}

		$('#tb_actionbar_item_userlist')
			.w2overlay({
				class: 'cool-font',
				html: message,
				style: 'padding: 5px'
			});
```

so it is sanitized with `escapeHtml` as soon as we get it, so I believe it not to be a vulnerability, but I agree it seems rather like it could become a problem with someone making a mistake. I'll clean it and the other bits up